Source/WebKit/UIProcess/WebPageProxy.cpp

```diff
@@ -8392,6 +8393,28 @@ void WebPageProxy::decidePolicyForResponseShared(Ref<WebProcessProxy>&& process,
         m_navigationClient->decidePolicyForNavigationResponse(*this, WTFMove(navigationResponse), WTFMove(listener));
 }
 
+void WebPageProxy::showBrowsingWarning(RefPtr<WebKit::BrowsingWarning>&& safeBrowsingWarning)
```

Comment:
Seems like you could factor out the existing completion handler into this one, too.